LayoutTests/imported/w3c/web-platform-tests/import-maps/no-referencing-script-integrity-valid.html

```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+let log = [];
+</script>
+<script type="importmap">
+{
+  "integrity": {
+    "./resources/log.js?pipe=sub&name=NoReferencingScriptValidCheck": "sha384-5eRmXQSBE6H5ENdymdZxcyiIfJL1dxtH8p+hOelZY7Jzk+gt0gYyemrGY0cEaThF"
+  }
+}
+</script>
+<script>
+let promiseResolve;
+let promiseReject;
+let promise = new Promise((resolve, reject) => {
+  promiseResolve = resolve;
+  promiseReject = reject;
+});
+</script>
+</head>
+<body>
+<img src="/images/green.png?2"
+  onload="import('./resources/log.js?pipe=sub&name=NoReferencingScriptValidCheck').then(promiseResolve).catch(promiseReject)">
+<script>
+promise_test(async () => {
+  await promise;
+  assert_equals(log.length, 1);
+  assert_equals(log[0], "log:NoReferencingScriptValidCheck");
+}, "Script was loaded as its valid integrity check passed");
+</script>
```

Comment:
Why not just: 

```suggestion
</head>
<body>
<img>
<script>
promise_test(async () => {
  const img = document.querySelector("img");
  await new Promise((resolve) => {
    img.src = "/images/green.png?2";
    img.onload = resolve;
  });
  await import('./resources/log.js?pipe=sub&name=NoReferencingScriptValidCheck');
  assert_equals(log.length, 1);
  assert_equals(log[0], "log:NoReferencingScriptValidCheck");
}, "Script was loaded as its valid integrity check passed");
</script>
```

Reply:
The test is testing that the import statement as part of an HTML-based onload event is working as expected (and has the right integrity metadata). So I don't think we want to move the import outside of the HTML based onload handler.

Reply:
ah! I see now. ok, makes sense... Maybe cleaner as?:

```JS
promise_test(async () => {
  const promise = new Promise((resolve, reject) => {
    document.getElementById('test-img').onload = () => {
      import('./resources/log.js?pipe=sub&name=NoReferencingScriptValidCheck').then(resolve).catch(reject);
    };
  });

  await promise;
  assert_equals(log.length, 1);
  assert_equals(log[0], "log:NoReferencingScriptValidCheck");
}, "Script was loaded as its valid integrity check passed");
```

Reply:
This is testing the part of [HostLoadImportedModule](https://html.spec.whatwg.org/multipage/webappapis.html#hostloadimportedmodule) where step 6's condition is false and `referencingScript` remains null. For that (IIUC), the onload event must be defined as an HTML attribute, outside of any script tag.

Reply:
I'll add a comment to that effect, as it's not necessarily clear